nucliadb/migrations/0017_purge_orphan_kbslugs.py

```diff
@@ -0,0 +1,55 @@
+# Copyright (C) 2021 Bosutech XXI S.L.
+#
+# nucliadb is offered under the AGPL v3.0 and as commercial software.
+# For commercial licensing, contact us at info@nuclia.com.
+#
+# AGPL:
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""Migration #17
+
+Due to a bug on backend services, some kbslugs were not properly deleted and got
+orphan. Let's delete them!
+
+"""
+import logging
+
+from nucliadb.ingest.orm.knowledgebox import KB_SLUGS_BASE, KnowledgeBox
+from nucliadb.migrator.context import ExecutionContext
+
+logger = logging.getLogger(__name__)
+
+
+async def migrate(context: ExecutionContext) -> None:
+    async with context.kv_driver.transaction() as txn:
+        async for key in txn.keys(KB_SLUGS_BASE, count=-1):
+            slug = key.replace(KB_SLUGS_BASE, "")
+            value = await txn.get(key)
+            if value is None:
```

Comment:
shouldn't we delete the key in this case too?

Reply:
Sure!